### PR TITLE
Add support for RESTEasy 6.x

### PR DESCRIPTION
**Pull request by Leibnizhu**
### Add an agent plugin to support RESTEasy 6.x

- [x] Add a test case for the new plugin, refer to [the doc](https://github.com/apache/skywalking-java/blob/main/docs/en/setup/service-agent/java-agent/Plugin-test.md)  
- [x] Add a component id in [the component-libraries.yml](https://github.com/apache/skywalking/blob/master/oap-server/server-starter/src/main/resources/component-libraries.yml)  
- [x] Add a logo in [the UI repo](https://github.com/apache/skywalking-rocketbot-ui/tree/master/src/views/components/topology/assets)  

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes : NO
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

Skywalking already supports Resteasy 3.x and 4.x. Referrers to [RESTEASY-3007](https://issues.redhat.com/browse/RESTEASY-3007), Resteasy 6 has migrated from Jakarta REST 2.1 to Jakarta REST 3.0.  
Therefore, when applying skywalking's Resteasy 4.x javaagent to a Resteasy 6.x application, it throws `NoSuchMethodError` like belows:

```
java.lang.NoSuchMethodError: 'javax.ws.rs.core.HttpHeaders org.jboss.resteasy.spi.HttpRequest.getHttpHeaders()'
	at org.apache.skywalking.apm.plugin.resteasy.v4.server.SynchronousDispatcherInterceptor.beforeMethod(SynchronousDispatcherInterceptor.java:50)
	at org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstMethodsInter.intercept(InstMethodsInter.java:76)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:240)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:154)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:321)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:157)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:229)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:82)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:147)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:93)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

this PR:

1. Add support for RESTEasy 6.x
2. Test skywalking's RESTEasy 4.x javaagent (#265 ) for application with RESTEasy 5.x framework (add 5.x versions in `support-version.list`)
3. Add RESTEasy 6.x tests into github workflow; according to [Apache Tomcat Versions](https://tomcat.apache.org/whichversion.html), we choose jdk17+tomcat10 for jakarta REST 3.0 supporting

### CODE REVIEW

File: .github/workflows/plugins-jdk17-test.1.yaml [hunk 1]
```diff
@@ -56,6 +56,7 @@ jobs:
       matrix:
         case:
           - spring-6.x-scenario
+          - resteasy-6.x-scenario
```
Comment (wu-sheng) [inline .github/workflows/plugins-jdk17-test.1.yaml:59]: JDK17 is resteasy running minimal requirement?

File: apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        <artifactId>resteasy-plugin</artifactId>
+        <groupId>org.apache.skywalking</groupId>
+        <version>9.0.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>resteasy-server-6.x-plugin</artifactId>
+    <packaging>jar</packaging>
+
+    <name>apm-resteasy-server-6.x-plugin</name>
+    <url>http://maven.apache.org</url>
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/pom.xml:32]: Why maven apache?

File: apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/resteasy/v6/server/SynchronousDispatcherExceptionInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                              Object ret) throws Throwable {
+        return ret;
+    }
+
+    @Override
+    public void handleMethodException(EnhancedInstance objInst, Method method, Object[] allArguments,
+                                      Class<?>[] argumentsTypes, Throwable t) {
+        ContextManager.activeSpan().log(t);
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/resteasy/v6/server/SynchronousDispatcherExceptionInterceptor.java:48]: In before, there is `ContextManager.isActive()`, it means there is a possibility that no-active-context. But in this case, you would be NPE.

Could you be more clear whether no-active-context exists or not?

File: .github/workflows/plugins-jdk17-test.1.yaml (hunk 1 above)
Comment (Leibnizhu) [reply .github/workflows/plugins-jdk17-test.1.yaml:59]: minimal requirement is tomcat 10 and jdk11, should i create a new github workflow for jdk11 tests?

Comment (wu-sheng) [reply .github/workflows/plugins-jdk17-test.1.yaml:59]: But your plugin is still compiled at JDK8 level only, right?

File: apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/pom.xml (hunk 2 above)
Comment (Leibnizhu) [reply apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/pom.xml:32]: Um, I'm not quite sure. I referenced other plugins' pom.xml.

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/pom.xml:32]: I don't think we need this.

Comment (Leibnizhu) [reply apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/pom.xml]: removed

File: apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/resteasy/v6/server/SynchronousDispatcherExceptionInterceptor.java (hunk 3 above)
Comment (Leibnizhu) [reply apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/resteasy/v6/server/SynchronousDispatcherExceptionInterceptor.java]: I think this is necessary defensive code.
code in `SynchronousDispatcherExceptionInterceptor` and `SynchronousDispatcherInterceptor` are adjusted.

File: .github/workflows/plugins-jdk17-test.1.yaml (hunk 1 above)
Comment (Leibnizhu) [reply .github/workflows/plugins-jdk17-test.1.yaml:59]: this plugin can be compiled at jdk8, and run with resteasy+tomcat in jdk11 (or later)

**Review by wu-sheng — APPROVED** [review]

**Comment by Leibnizhu** [comment]
@wu-sheng btw, when will skywalking 9.0.0 be released?

**Comment by wu-sheng** [comment]
At least end of next month, the last 8.x released last month.

File: apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/resteasy/v6/server/define/SynchronousDispatcherInstrumentation.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+import org.apache.skywalking.apm.agent.core.plugin.match.NameMatch;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+public class SynchronousDispatcherInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+
+    private static final String ENHANCE_CLASS = "org.jboss.resteasy.core.SynchronousDispatcher";
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/resteasy/v6/server/define/SynchronousDispatcherInstrumentation.java:34]: @Leibnizhu We received a report from slack that all v3/v4/v6 plugins of RESTEasy are using the same target class without witness class. It seems to have conflicts.

Comment (Leibnizhu) [reply apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/resteasy/v6/server/define/SynchronousDispatcherInstrumentation.java:34]: Where can I find more information related to `witness class` ? or in other words, how should I fix it?

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/resteasy/v6/server/define/SynchronousDispatcherInstrumentation.java:34]: https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/java-plugin-development-guide/#implement-plugin

Read point 4 here.

Witness class is using version specific class names or methods to separate versions in the runtime.

Comment (Leibnizhu) [reply apm-sniffer/apm-sdk-plugin/resteasy-plugin/resteasy-server-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/resteasy/v6/server/define/SynchronousDispatcherInstrumentation.java:34]: I fixed it in PR #722

**Pull request merged**